cmd/spot/main.go

```diff
@@ -198,7 +200,10 @@ func runAdHoc(ctx context.Context, targets []string, r *runner.Process) error {
 
 // runGen generates a destination report for the task's targets
 func runGen(opts options, r *runner.Process) (err error) {
-	targets := targetsForTask(opts.Targets, opts.TaskName, r.Playbook)
+	var targets []string
+	for _, taskName := range opts.TaskNames {
```

Comment:
probably this will cause duplicated  targets in the final list. I'm not sure how bad the end result is going to be but deduplication of the result slice sounds like the right thing to do

Reply:
Actually, it causes duplicates in the targets list. I've updated the code and added a test to cover the case.